syntax = "proto3";

package user;

service UserService {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
  rpc AuthenticateUser (AuthenticateUserRequest) returns (AuthenticateUserResponse);
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
  string password = 3;
}

message CreateUserResponse {
  string user_id = 1;
  string message = 2;
}

message AuthenticateUserRequest {
  string email = 1;
  string password = 2;
}

message AuthenticateUserResponse {
  string token = 1;
  string message = 2;
}
